test/messaging.test.ts

```diff
@@ -0,0 +1,106 @@
+import { solidity } from "ethereum-waffle";
+import chai, { assert, expect } from "chai";
+import { simpleDeploy } from "@makerdao/hardhat-utils";
+import hre, { ethers, network, starknet } from "hardhat";
+import { StarknetContract, HardhatUserConfig } from "hardhat/types";
+import config from "../hardhat.config";
+
+describe("test contracts interaction", function () {
+
+    it("check deployment data", async () => {
+        const { l1Contract, l2Contract, l2user, mockStarknetMessagingAddress, admin, l1_user } = await setupTest();
```

Comment:
you can define these variables once and use `before` block instead of calling the same function `setupTest()` at the beginning of `it` blocks. see:

```
import { Contract } from "ethers";
import { StarknetContract, Account } from "hardhat/types";
import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";


describe("test contracts interaction", function () {
    let l1Contract: Contract; 
    let l2Contract: StarknetContract;
    let l2user: Account;
    let mockStarknetMessagingAddress: string;
    let admin: SignerWithAddress;
    let l1_user: SignerWithAddress;

    before(async function () {
        let { l1Contract, l2Contract, l2user, mockStarknetMessagingAddress, admin, l1_user } = await setupTest();
    })
    
    it("check deployment data", async () => {
        ...
    })
})
```